### PR TITLE
parse: Fix DIR_SECURITY data directory retrieval

### LINKED ISSUE #121: GetDataDirectoryEntry returns an incorrect buffer for DIR_SECURITY
`GetDataDirectory` currently assumes that each data directory vector entry's `VirtualAddress` corresponds to a (materializable) RVA. This is true for all known entries **except** `DIR_SECURITY`, which is a special case. Per MSDN:

> The Certificate Table entry points to a table of attribute certificates. These certificates are not loaded into memory as part of the image. As such, the first field of this entry, which is normally an RVA, is a file pointer instead.

`GetDataDirectoryEntry` needs to special-case `DIR_SECURITY` and grab the buffer by file offset instead of by section + RVA lookup.

### PR DESCRIPTION
**Pull request by woodruffw**
WIP until I test it.

Fixes #121.

### CODE REVIEW

File: pe-parser-library/src/parse.cpp [hunk 1]
```diff
@@ -2655,20 +2655,35 @@ bool GetDataDirectoryEntry(parsed_pe *pe,
...
 
-  section sec;
-  if (!getSecForVA(pe->internal->secs, addr, sec)) {
-    PE_ERR(PEERR_SECTVA);
-    return false;
-  }
+  /* NOTE(ww): DIR_SECURITY is an annoying special case: its contents
+   * are never mapped into memory, so its "RVA" is actually a direct
+   * file offset.
+   */
```
Comment (ekilmer) [inline pe-parser-library/src/parse.cpp]: I think it would be useful to paste a URL with a source for this comment

Comment (ekilmer) [reply pe-parser-library/src/parse.cpp]: I think it should be this link, right? https://docs.microsoft.com/en-us/windows/win32/debug/pe-format#the-attribute-certificate-table-image-only

Comment (woodruffw) [reply pe-parser-library/src/parse.cpp]: Yep, that looks right. Will add.

**Comment by woodruffw** [comment]
> Eventually, we'll want to add some test cases for special cases like this.

Yeah, we're sorely in need of tests here. I was thinking that it might be good to have a stand-alone repo that produces a bunch of PEs in different configurations, which we could then expose as an independent GitHub Action. That way we could reuse them across multiple projects (e.g., winchecksec).

**Comment by ekilmer** [comment]
> I was thinking that it might be good to have a stand-alone repo that produces a bunch of PEs in different configurations, which we could then expose as an independent GitHub Action. That way we could reuse them across multiple projects (e.g., winchecksec).

I like this idea. We can talk more about how testing should be setup in another thread, but we should make sure that people can easily test against a separate test repo locally, using submodules or something.

**Pull request merged**